stdlib/2and3/contextlib.pyi

```diff
@@ -64,3 +64,29 @@ if sys.version_info >= (3,):
         def pop_all(self) -> ExitStack: ...
         def close(self) -> None: ...
         def __enter__(self: _U) -> _U: ...
+
+if sys.version_info >= (3, 7):
+    from typing import Awaitable
+
+    _U = TypeVar('_U', bound='AsyncExitStack')
+
+    _ExitCoroFunc = Callable[[Optional[Type[BaseException]],
+                              Optional[BaseException],
+                              Optional[TracebackType]], Awaitable[bool]]
+    _CallbackCoroFunc = Callable[..., Awaitable[None]]
+    _ACM_EF = TypeVar('_ACM_EF', AsyncContextManager, _ExitCoroFunc)
+
+    class AsyncExitStack(AsyncContextManager[AsyncExitStack]):
+        def __init__(self) -> None: ...
+        def enter_context(self, cm: ContextManager[_T]) -> _T: ...
+        def enter_async_context(self, cm: AsyncContextManager[_T]) -> Awaitable[_T]: ...
+        def push(self, exit: _CM_EF) -> _CM_EF: ...
+        def push_async_exit(self, exit: _ACM_EF) -> _ACM_EF: ...
+        def callback(self, callback: Callable[..., None],
+                     *args: Any, **kwds: Any) -> Callable[..., None]: ...
+        def push_async_callback(self, callback: _CallbackCoroFunc,
+                                *args: Any, **kwds: Any) -> _CallbackCoroFunc: ...
+        def pop_all(self) -> ExitStack: ...
```

Comment:
Should return AsyncExitStack.

Reply:
Actually, I think it should be implemented as `__enter__` / `__aenter__`: `pop_all` [instantiates](https://github.com/python/cpython/blob/master/Lib/contextlib.py#L391) `type(self)`.

Probably that should also be fixed for `ExitStack`.

Reply:
Thanks! Would you mind fixing it there too while you're at it?